src/main/java/sg/ncl/MainController.java

```diff
@@ -602,83 +624,25 @@ public String editAccountDetails(@ModelAttribute("editUser") User2 editUser, fin
                 credHeaders.setContentType(MediaType.APPLICATION_JSON);
                 credHeaders.set("Authorization", AUTHORIZATION_HEADER);
 
-                HttpEntity<String> credRequest = new HttpEntity<String>(credObject.toString(), headers);
-                restTemplate.exchange(properties.getSioCredUrl() + session.getAttribute("id"), HttpMethod.PUT, credRequest, String.class);
-                redirectAttributes.addFlashAttribute("editPassword", "success");
+                HttpEntity<String> credRequest = new HttpEntity<>(credObject.toString(), headers);
+                restTemplate.setErrorHandler(new MyResponseErrorHandler());
+                ResponseEntity response = restTemplate.exchange(properties.getSioCredUrl() + session.getAttribute("id"), HttpMethod.PUT, credRequest, String.class);
+                String responseBody = response.getBody().toString();
+
+                try {
+                    if (RestUtil.isError(response.getStatusCode())) {
+                        MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                        System.out.println(error.getExceptionName());
+                        redirectAttributes.addFlashAttribute("editPassword", "fail");
+                    } else {
+                        redirectAttributes.addFlashAttribute("editPassword", "success");
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)